AlgorithmContext.h

```diff
@@ -27,6 +28,8 @@ class AlgorithmContext {
   void run_algorithm();
 
 private:
+  std::string context_name;
+  InputOutput<std::vector> &io;

```

Comment:
reference in parent object 'require' child objects life to be at least as long as parent's. are we sure IO is alive that long?

Reply:
Both created in `main.cpp` in a single loop iteration and both have automatic storage duration within the loop code block.